src/dfi/consensus/txvisitor.cpp

```diff
@@ -96,19 +97,17 @@ CCustomTxVisitor::CCustomTxVisitor(const CTransaction &tx,
                                    const Consensus::Params &consensus,
                                    const uint64_t time,
                                    const uint32_t txn,
-                                   const std::shared_ptr<CScopedQueueID> &evmQueueId,
-                                   const bool isEvmEnabledForBlock,
-                                   const bool evmPreValidate)
+                                   const BlockContext& blockCtx)
     : height(height),
       mnview(mnview),
       tx(tx),
       coins(coins),
       consensus(consensus),
       time(time),
       txn(txn),
-      evmQueueId(evmQueueId),
-      isEvmEnabledForBlock(isEvmEnabledForBlock),
-      evmPreValidate(evmPreValidate) {}
+      evmQueueId(blockCtx.evmQueueId),
+      isEvmEnabledForBlock(blockCtx.isEvmEnabledForBlock),
+      evmPreValidate(blockCtx.evmPreValidate) {}
```

Comment:
evmPrevalidate would be tx context, and not block context?

Reply:
At what point do we switch to a transaction context?